src/redux/services/validation.ts

```diff
@@ -49,149 +47,151 @@ export function validateResource(resource: K8sResource, schemaVersion: string, u
+          }
           return error;
         })
-      );
-    }
+    );
+  }
 
-    if (!validatorCache.has(validatorCacheKey)) {
-      const ajv = new Ajv({
-        unknownFormats: 'ignore',
-        validateSchema: false,
-        logger: false,
-        jsonPointers: true,
-        verbose: true,
-        allErrors: true,
-      });
-      validatorCache.set(validatorCacheKey, ajv.compile(resourceSchema));
-    }
+  errors.sort(compareErrorsByLineNumber);
 
-    const validate = validatorCache.get(validatorCacheKey);
-
-    if (validate) {
-      try {
-        validate(resource.content);
-
-        if (validate.errors) {
-          errors.push(
-            ...validate.errors
-              .filter(
-                err =>
-                  !ignoredProperties.some(
-                    ignored =>
-                      (ignored.startsWith('*') && err.dataPath.endsWith(ignored.substring(1))) ||
-                      (ignored.endsWith('*') && err.dataPath.startsWith(ignored.substring(0, ignored.length - 1))) ||
-                      err.dataPath === ignored
-                  )
-              )
-              .map(err => {
-                const parsedDoc = getParsedDoc(resource);
-
-                // @ts-ignore
-                const valueNode = findJsonPointerNode(parsedDoc, err.dataPath.substring(1).split('/'));
-
-                const error: ResourceValidationError = {
-                  property: err.dataPath,
-                  message: err.message || 'message',
-                  errorPos: valueNode ? getErrorPosition(valueNode, getLineCounter(resource)) : undefined,
-                  // @ts-ignore
-                  description: err.parentSchema?.description || undefined,
-                };
-
-                // @ts-ignore
-                if (err.keyword === 'additionalProperties' && err.params?.additionalProperty) {
-                  // @ts-ignore
-                  error.message += `: ${err.params.additionalProperty}`;
-                }
-                return error;
-              })
-          );
-        }
-
-        // sort errors by line number
-        errors.sort((e1, e2): number => {
-          if (e1.errorPos && !e2.errorPos) {
-            return 1;
-          }
-          if (!e1.errorPos && e2.errorPos) {
-            return -1;
-          }
-          if (!e1.errorPos && !e2.errorPos) {
-            return 0;
-          }
+  resource.validation = {
+    isValid: errors.length === 0,
+    errors,
+  };
+}
 
-          // @ts-ignore
-          return e1.errorPos.line - e2.errorPos.line;
-        });
+function lazyGetResourceValidator(
+  resource: K8sResource,
+  schemaVersion: string,
+  userDataDir: string
+): Ajv.ValidateFunction | undefined {
+  const resourceSchema = getResourceSchema(resource, schemaVersion, userDataDir);
 
-        resource.validation = {
-          isValid: errors.length === 0,
-          errors: errors && errors.length > 0 ? errors : [],
-        };
-      } catch (e) {
```

Comment:
This try-catch should be unnecessary as seen from the AJV docs. Please double check if I'm overlooking why this should throw.

> Compiled parsers do NOT throw exceptions
>
> Compiled parsers, unlike JSON.parse, do not throw the exception in case JSON string is not a valid JSON or in case data is invalid according to the schema. As soon as the parser determines that either JSON or data is invalid, it returns undefined and reports error and position via parsers properties message and position.

Reply:
Thanks @WitoDelnat , this is indeed the case so the try catch is not needed